src/main/java/org/opentripplanner/routing/impl/DefaultFareServiceImpl.java

```diff
@@ -101,14 +102,15 @@ public void addFareRules(FareType fareType, Collection<FareRuleSet> fareRules) {
             if ( ! (edge instanceof HopEdge))
                 continue;
             HopEdge hEdge = (HopEdge) edge;
-            if (ride == null || ! state.getRoute().equals(ride.route)) {
+            if (ride == null || ! state.getRoute().equals(ride.route) || createRideDueToInterlining(state, ride)) {
```

Comment:
Could you add a comment explaining what this conditional block is doing?

Reply:
Comment added in 7d5e6d4cf01b5e8212d232d4281329328d26d7f7.